kafka-rest/src/main/java/io/confluent/kafkarest/ratelimit/FixedCostRateLimitRequestFilter.java

```diff
@@ -15,26 +15,43 @@
 
 package io.confluent.kafkarest.ratelimit;
 
+import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
+import com.google.common.cache.LoadingCache;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
 
 /**
  * A {@link ContainerRequestFilter} that automatically applies a request rate-limit at a fixed cost
- * based on the resource/method being requested, according to the {@link
+ * based on the resource/method being requested per cluster, according to the {@link
  * io.confluent.kafkarest.KafkaRestConfig#RATE_LIMIT_COSTS_CONFIG} and {@link
  * io.confluent.kafkarest.KafkaRestConfig#RATE_LIMIT_DEFAULT_COST_CONFIG} configs.
  */
 final class FixedCostRateLimitRequestFilter implements ContainerRequestFilter {
-  private final FixedCostRateLimiter rateLimiter;
+  private final RequestRateLimiter genericRateLimiter;
+  private final int cost;
+  private final LoadingCache<String, RequestRateLimiter> perClusterRateLimiterCache;
 
-  FixedCostRateLimitRequestFilter(FixedCostRateLimiter rateLimiter) {
-    this.rateLimiter = requireNonNull(rateLimiter);
+  FixedCostRateLimitRequestFilter(
+      RequestRateLimiter genericRateLimiter,
+      int cost,
+      LoadingCache<String, RequestRateLimiter> perClusterRateLimiterCache) {
+    checkArgument(cost > 0, "Cost must be positive");
+    this.genericRateLimiter = requireNonNull(genericRateLimiter);
+    this.cost = cost;
+    this.perClusterRateLimiterCache = requireNonNull(perClusterRateLimiterCache);
   }
 
   @Override
   public void filter(ContainerRequestContext requestContext) {
-    rateLimiter.rateLimit();
+    String clusterId = requestContext.getUriInfo().getPathParameters(true).getFirst("clusterId");
+    if (clusterId != null) {
+      RequestRateLimiter rateLimiter = perClusterRateLimiterCache.getUnchecked(clusterId);
+      rateLimiter.rateLimit(cost);
```

Comment:
I think we still need to generic rate limit here too.

We have eg a global total of 25 requests/s

If we have 100 different LKCs all hitting us with eg 10 requests/s.  We'll blow the global limit.  

I know the dos filter does 25 requests/s so this is a bit redundant atm, but I see the dos limiter going up to eg 100 requests/s soon, and this generic limiter limiting admin requests to 25/s and the produce limiters allowing the other 75/s

Reply:
this make sense 👍 I'll adjust it accordingly.

Reply:
@trnguyencflt i think this should be documented in a comment. Better yet maybe add a wiki on latest state of rate-limiting in kREST(global-limits, per lkc limits, per ip limits, admin vs produce limits etc.), then u can link that in the comment. That wiki could also link to important places in code where this all setup. That should help with the review of this PR as well :) What do you think?

PS - I do see a [wiki](https://confluentinc.atlassian.net/wiki/spaces/KREST/pages/2939096773/Rate+Limiting) exists, maybe u can update that.

Reply:
@msn-tldr thanks for the comment, that work is planned in this ticket https://confluentinc.atlassian.net/browse/KREST-8608 😄 

As this PR is not yet finalised, so I would delay the documentation update until this is merged.